python/cuml/benchmark/algorithms.py

```diff
@@ -97,10 +101,15 @@ def run_cpu(self, data, **override_args):
         if self.data_prep_hook:
             data = self.data_prep_hook(data)
         if self.accepts_labels:
-            cpu_obj.fit(data[0], data[1])
+            if self.bench_func is not None:
```

Comment:
the nesting was a little confusing here... how about starting with:

`bench_func = self.bench_func if self.benc_func is not None else cpu_obj.fit`

then get rid of the inner branch and always call `bench_func(cpu_obj, ...)`

Reply:
The problem with that is the way need to be invoked is different, so I can't assign them both to `bench_func`. This just gave me an idea- perhaps a bench function should always be used, and should just default to a `fit_function`. That allows us to clean up this call, and makes it extensible by design.